src/sql/workbench/parts/notebook/browser/models/notebookInput.ts

```diff
@@ -38,7 +39,7 @@ export class NotebookEditorModel extends EditorModel {
 	private readonly _onDidChangeDirty: Emitter<void> = this._register(new Emitter<void>());
 	private _lastEditFullReplacement: boolean;
 	constructor(public readonly notebookUri: URI,
-		private textEditorModel: TextFileEditorModel | UntitledEditorModel,
+		private textEditorModel: TextFileEditorModel | UntitledEditorModel | ResourceEditorModel,
```

Comment:
Any thoughts on how I can make this cleaner throughout? Can I reference all of these as a type or something?

Reply:
yes, you can do this. export type TextEditorModel = TextFileEditorModel | UntitledEditorModel | ResourceEditorModel;

Reply:
You could just have the type be `BaseTextEditorModel`

Reply:
From what I can tell text file, untitled and resource editor are the only classes that extend it.

Reply:
Thanks. I cleaned this up everywhere else (except where the logic got really hairy... like right here. I'd rather leave this for now 😄)